Source/WebCore/page/LocalFrameViewLayoutContext.cpp

```diff
@@ -621,7 +621,22 @@ bool LocalFrameViewLayoutContext::pushLayoutState(RenderBox& renderer, const Lay
     
 void LocalFrameViewLayoutContext::popLayoutState()
 {
+    if (!layoutState())
+        return;
+
+    auto currentLineClamp = layoutState()->lineClamp();
+
     m_layoutStateStack.removeLast();
+
+    if (currentLineClamp) {
+        // Propagates the current line clamp state to the parent.
+        if (auto* layoutState = this->layoutState(); layoutState->lineClamp()) {
+            layoutState->setLineClamp(std::make_optional(RenderLayoutState::LineClamp {
```

Comment:
make_optional can be omitted
`layoutState->setLineClamp(RenderLayoutState::LineClamp {`